engine/src/main/java/org/pentaho/di/base/AbstractBaseCommandExecutor.java

```diff
@@ -234,6 +237,28 @@ public boolean isEnabled( final String value ) {
     return YES.equalsIgnoreCase( value ) || Boolean.parseBoolean( value ); // both are NPE safe, both are case-insensitive
   }
 
+  /**
+   * Decodes the provided base64String into the specified filePath. Parent directories must already exist.
+   *
+   * @param base64String String BASE64 representation of a file
+   * @param filePath     String The path to which the base64String is to be decoded
+   * @return File the newly created File
+   */
+  protected File decodeBase64StringToFile( String base64String, String filePath ) {
+    //Decode base64String to byte[]
+    byte[] decodedBytes = Base64.getDecoder().decode( base64String );
+    File file = new File( filePath );
+
+    //Write to file
+    try ( FileOutputStream fos = new FileOutputStream( file ) ) {
+      fos.write( decodedBytes );
+    } catch ( Exception e ) {
+      getLog().logError( " Exception writing decoded BASE64 string to file " + e.toString() );
```

Comment:
One of 2 things should occur here:
 - Opt A) throw the exception ( add it to method signature ); not the generic `Exception`, but the `IOException` one
 - Opt B) set file to `null`, so it ends up getting returned as a `null` object, rather than a corrupted one

If none of the above occur, and in a scenario where some exception takes please, the person leveraging this method would never know of it, and rely that the returned `File` object is a solid one.

Reply:
I will go with A